src/libasr/asr_utils.h

```diff
@@ -1912,6 +1912,17 @@ static inline void get_dimensions(ASR::expr_t* array, Vec<ASR::expr_t*>& dims,
     }
 }
 
+static inline ASR::EnumType_t* get_EnumType_from_symbol(ASR::symbol_t* s) {
+    ASR::Variable_t* s_var = ASR::down_cast<ASR::Variable_t>(s);
+    if( ASR::is_a<ASR::Enum_t>(*s_var->m_type) ) {
+        ASR::Enum_t* enum_ = ASR::down_cast<ASR::Enum_t>(s_var->m_type);
+        return ASR::down_cast<ASR::EnumType_t>(enum_->m_enum_type);
+    }
+    ASR::symbol_t* enum_type_cand = ASR::down_cast<ASR::symbol_t>(s_var->m_parent_symtab->asr_owner);
+    LFORTRAN_ASSERT(ASR::is_a<ASR::EnumType_t>(*enum_type_cand));
+    return ASR::down_cast<ASR::EnumType_t>(enum_type_cand);
```

Comment:
This PR is fine. Keeping the `type` in `EnumName/Value` would simplify this function, but otherwise it seems it's not needed that much.

I still don't like how complicated this function is, it feels like we are lumping together some unrelated things.